### PR TITLE
Eth2_hashing: Downgrade `ethereum-types` and revert changes

### PR DESCRIPTION
**Pull request by karim-en**
* Revert changes of `eth2_hashing`
* Downgrade `ethereum-types` to "0.9.2"

### CODE REVIEW

**Review by sept-en — APPROVED** [review]

File: consensus/tree_hash/src/lib.rs [hunk 1]
```diff
@@ -7,7 +7,7 @@ pub use merkle_hasher::{Error, MerkleHasher};
 pub use merkleize_padded::merkleize_padded;
 pub use merkleize_standard::merkleize_standard;
 
-use eth2_hashing::{hash_fixed, ZERO_HASHES, ZERO_HASHES_MAX_INDEX};
+use eth2_hashing::{hash32_concat, hash_fixed, ZERO_HASHES, ZERO_HASHES_MAX_INDEX};
```
Comment (sept-en) [inline consensus/tree_hash/src/lib.rs]: Please revert these changes too (will be easier to maintain with an upstream)

**Review by sept-en — CHANGES_REQUESTED** [review]

File: consensus/tree_hash/src/lib.rs (hunk 1 above)
Comment (karim-en) [reply consensus/tree_hash/src/lib.rs]: 04a537f5a7c593f37dcfa80493b36b363bf29daf
